examples/ui-demo/src/components/small-cards/SolanaNftCard.tsx

```diff
@@ -0,0 +1,228 @@
+import { useToast } from "@/hooks/useToast";
+import { useSolanaTransaction } from "@account-kit/react";
+import {
+  Keypair,
+  PublicKey,
+  SystemProgram,
+  VersionedTransaction,
+} from "@solana/web3.js";
+
+import {
+  ExtensionType,
+  LENGTH_SIZE,
+  TOKEN_2022_PROGRAM_ID,
+  TYPE_SIZE,
+  createInitializeInstruction,
+  createInitializeMetadataPointerInstruction,
+  createInitializeMintInstruction,
+  createUpdateFieldInstruction,
+  getMintLen,
+} from "@solana/spl-token";
+import { pack, TokenMetadata } from "@solana/spl-token-metadata";
+import { LoadingIcon } from "../icons/loading";
+import { Button } from "./Button";
+import { Card } from "./Card";
+import Image from "next/image";
+import { Badge } from "./Badge";
+import { CheckCircleFilledIcon } from "../icons/check-circle-filled";
+import { useState } from "react";
+
+type TransactionState = "idle" | "signing" | "sponsoring" | "complete";
+
+export const SolanaNftCard = () => {
+  const { setToast } = useToast();
+  const {
+    sendTransactionAsync,
+    isPending,
+    connection,
+    signer: solanaSigner,
+  } = useSolanaTransaction({
+    mutation: {
+      onError(error) {
+        console.log(error);
+        setToast({
+          type: "error",
+          text: "Error sending transaction",
+          open: true,
+        });
+      },
+    },
+  });
+  const [transactionState, setTransactionState] =
+    useState<TransactionState>("idle");
+
+  const handleCollectNFT = async () => {
+    if (!solanaSigner) throw new Error("No signer found");
+    if (!connection) throw new Error("No connection found");
+    setTransactionState("signing");
+    setTransactionState("sponsoring");
+    const stakeAccount = Keypair.generate();
+    const publicKey = new PublicKey(solanaSigner.address);
+    const metaData: (readonly [string, string])[] = [
+      ["Background", "Blue"],
+      ["WrongData", "DeleteMe!"],
+      ["Points", "0"],
+    ];
+    const tokenMetadata: TokenMetadata = {
+      updateAuthority: publicKey,
+      mint: stakeAccount.publicKey,
+      name: "Alchemy Duck",
+      symbol: "ALCHDUCK",
+      uri: "https://bafybeigtvzjqalevyw67xdhr7am5r3jxe5kjbg4pi2jv3nxvhelptwksoe.ipfs.dweb.link?filename=duckImage.png",
+      additionalMetadata: metaData,
+    };
+    const decimals = 6;
+    const mintLen = getMintLen([ExtensionType.MetadataPointer]);
+    const metadataLen = TYPE_SIZE + LENGTH_SIZE + pack(tokenMetadata).length;
+    const mintLamports = await connection.getMinimumBalanceForRentExemption(
+      mintLen + metadataLen
+    );
+
+    const mint = stakeAccount.publicKey;
+    const tx = await sendTransactionAsync({
```

Comment:
nit: I'm not a big fan of mixing using the async function with the callbacks in the hook (I know we do this other places in this demo app so not an issue).  If using the async function I prefer to wrap it in a try/catch and handle the error there.  In this example, the onError callback will be called if there is an error but we will still get an unhandled error in this function.

Reply:
Same, I was told to implement useMutation.

Reply:
But in this case, this is a middleware that one can add. Ad we expect a transform, like map
And the mutation is the mutation pattern since we use it https://tanstack.com/query/v4/docs/framework/react/reference/useMutation. 
As for the onError, sure, in this case it makes alot more sense to use a try catch